### PR TITLE
Fix build for go1.16

### PR DESCRIPTION
**Pull request by r0x0d**
* Fixed the dependencies for building the code against go1.16
* Added go1.16 as a matrix target for the unit tests
* Updated go.sum to have the correct version of the dependencies

### CODE REVIEW

**Comment by r0x0d** [comment]
pre-commit.ci rerun

File: go.mod [hunk 1]
```diff
@@ -1,20 +1,11 @@
...
 
-go 1.20
+go 1.16
 
 require (
 	git.sr.ht/~spc/go-log v0.0.0-20230531172318-1397be06f5f4
 	github.com/google/uuid v1.3.0
 	github.com/redhatinsights/yggdrasil v0.0.0-20210811162724-41397343c25b
-	google.golang.org/grpc v1.56.2
+	google.golang.org/grpc v1.51.0
```
Comment (r0x0d) [inline go.mod:9]: The version here needs to be `v1.51.0` as this is the latest one that works for us under RHEL 7. After that, `v1.52.0+` the grpc dependency will depend on a newer version of `golang.org/x/sys`, which causes the code to break on go1.16 (the latest version for RHEL7)

https://github.com/grpc/grpc-go/blob/v1.51.0/go.mod#L16

Comment (andywaltlova) [reply go.mod:9]: ok, makes sense

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
           go-version: ${{ matrix.go-version }}
 
       - name: Go build (source)
-        run: make
+        run: make build
 
       - name: Go build (tests)
```
Comment (andywaltlova) [inline .github/workflows/tests.yml:27]: Do we want to run tests in pipeline for both versions? 1.16 and 1.20? The original task for unit tests required tests to be run in container, right now the make command has image with 1.20 golang hard coded, it is necessary to change that if we want to run tests for 1.16

**Review by andywaltlova — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (r0x0d) [reply .github/workflows/tests.yml:27]: Ah, good observation! I would say yes, that would be nice to have. 

I will make the changes to the container.

Comment (r0x0d) [reply .github/workflows/tests.yml]: done!

**Pull request merged**